cmd/containerd-stargz-grpc/main.go

```diff
@@ -119,12 +121,31 @@ func main() {
 			log.G(ctx).WithError(err).Fatalf("error on serving via socket %q", *address)
 		}
 	}()
-	waitForSIGINT()
-	log.G(ctx).Info("Got SIGINT")
+
+	if os.Getenv("NOTIFY_SOCKET") != "" {
+		notified, notifyErr := sddaemon.SdNotify(false, sddaemon.SdNotifyReady)
+		log.G(ctx).Debugf("SdNotifyReady notified=%v, err=%v", notified, notifyErr)
+	}
+
+	s := waitForSignal(unix.SIGINT, unix.SIGTERM)
+	log.G(ctx).Infof("Got %v", s)
+
+	if os.Getenv("NOTIFY_SOCKET") != "" {
+		notified, notifyErr := sddaemon.SdNotify(false, sddaemon.SdNotifyStopping)
+		log.G(ctx).Debugf("SdNotifyStopping notified=%v, err=%v", notified, notifyErr)
+	}
+	switch s {
+	case unix.SIGINT:
+		// do cleanup (defer) on SIGINT
+	default:
+		// exit
+		log.G(ctx).Info("Exiting")
+		os.Exit(0)
```

Comment:
Do we really need this?

Reply:
We need this for avoiding normal overlayfs (non-lazy) layers from being removed during `*snapshot.Close()`.

Reply:
What does os.Exit do for that?

Reply:
This avoids `defer` to be called (https://golang.org/pkg/os/#Exit).

Reply:
Ah, didn't know that. Please add comment lines for that. Or ideally we should disable the defer without depending on os.Exit.